crates/pathfinder/src/rpc/api.rs

```diff
@@ -62,7 +62,11 @@ impl RpcApi {
         Self {
             storage,
             sequencer,
-            chain,
+            chain_id: super::serde::bytes_to_hex_str(match chain {
+                Chain::Goerli => b"SN_GOERLI",
+                Chain::Mainnet => b"SN_MAIN",
+            })
```

Comment:
This does confuse me and I kind of think that we should avoid `String` whenever possible... Did you consider having a serialize impl for this? It could even select one out of two static (or maybe const) strings?

Reply:
A trivial solution is to select one of two hardcoded hex strs instead. The dynamic conversion would only be used for the expected value in the test.